litebird_sim/bandpasses.py

```diff
@@ -0,0 +1,275 @@
+from scipy import signal
+
+from typing import Union
+from uuid import UUID
+
+from .imo import Imo
+
+
+@dataclass
+class BandPassInfo(object):
+
+    """A class wrapping the basic information about a detector bandpass.
+    This  class   encodes the basic properties of a frequency band.
+    It can be initialized in three ways:
+    - Through the default constructor, will assume top-hat bandpasses
+    provided the band centroid and the width are issued.
+    - Through the class method :meth:`.from_imo`, which reads the
+      detector bandpass from the LiteBIRD Instrument Model (see
+      the :class:`.Imo` class).
+
+    Args:
+        - nu_center (float):  center frequency in GHz
+        - bandwidth (float):  width of the band (default=0.2)
+        - range_offset (int): increase the freq. range by an offset (default +- 10 from the edges)
+        - nsamp (int): number of elements to sample the band (default=128)
+        - name (str) : ID of the band
+        - normalize(bool) : If set to true bandpass weights will be normalized to 1
+        - bandtype (str): choose between:
+            - "top-hat" (default)
+            - "top-hat-exp" : the edges of the band are apodized with an exponential profile
+            - "top-hat-cosine" : the edges of the band are apodized with a cosine profile
+            - "cheby" : the bandpass encodes a _Chebyshev_  profile
+    """
+
+    bandcenter_ghz: float = 0.0
+    bandwidth_ghz: float = 0.0
+    range_offset: int = 10
+    nsamp: int = 128
+    name: str = ""
+    bandtype: str = "top-hat"
+    normalize: bool = False
+
+    def __post_init__(self):
+        """
+        Constructor of the class
+
+        """
+
+        self.f0, self.f1 = self.get_edges()
+
+        # we extend the wings of the top-hat bandpass  with 10 samples
+        # before  and after the edges
+        bandrange = self.f0 - range_offset, self.f1 + range_offset
+        self.freqs = np.linspace(bandrange[0], bandrange[1], nsamp)
+        self.isnormalized = False
+        if self.bandtype == "top-hat":
+            self.get_top_hat_bandpass(normalize=self.normalize)
+        elif self.bandtype == "top-hat-cosine":
+            self.get_top_hat_bandpass(apodization="cosine", normalize=self.normalize)
+        elif self.bandtype == "top-hat-exp":
+            self.get_top_hat_bandpass(apodization="exp", normalize=self.normalize)
+        elif self.bandtype == "cheby":
+            self.get_chebyshev_bandpass(normalize=self.normalize)
+        else:
+            print(f"{self.bandtype} profile not implemented. Assume top-hat")
+            self.get_top_hat_bandpass(normalize=self.normalize)
+
+    def get_edges(self):
+        """
+        get the edges of the tophat band
+        """
+        return (
+            self.bandcenter_ghz * (1 - self.bandwidth_ghz / 2),
+            self.bandcenter_ghz * (1 + self.bandwidth_ghz / 2),
+        )
+
+    def get_top_hat_bandpass(self, normalize=False, apodization=None):
```

Comment:
Is this function supposed to be called outside of the class? If not, better to rename it to `_get_top_hat_bandpass`, so that users are not tempted to use it.

Reply:
It might, but for time being i will make it private yes..